http://www.opengroup.org/onlinepubs/9699919799/functions/environ.html

--- origsrc/libssh-0.4.4/libssh/socket.c	2010-05-31 06:58:51.000000000 -0500
+++ src/libssh-0.4.4/libssh/socket.c	2010-06-01 13:57:38.947385300 -0500
@@ -33,7 +33,7 @@
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <sys/un.h>
-extern const char **environ;
+extern char **environ;
 #endif
 #include "libssh/priv.h"
 #include "libssh/socket.h"
