--- origsrc/libssh-0.2/Makefile.am	1969-12-31 18:00:00.000000000 -0600
+++ src/libssh-0.2/Makefile.am	2007-01-12 00:06:15.531250000 -0600
@@ -0,0 +1,13 @@
+SUBDIRS = include libssh
+
+INCLUDES = -Wall -g -I$(top_builddir)/include -I$(top_srcdir)/include
+
+noinst_PROGRAMS = samplessh samplesshd
+
+samplessh_SOURCES = sample.c
+
+samplesshd_SOURCES = samplesshd.c
+
+LDADD = libssh/libssh.la
+
+EXTRA_DIST = doc
--- origsrc/libssh-0.2/configure.in	2006-12-29 09:03:59.000000000 -0600
+++ src/libssh-0.2/configure.in	2007-01-12 00:09:47.468750000 -0600
@@ -5,6 +5,7 @@
 AC_INIT(libssh, 0.2 , aris@0xbadc0de.be)
 AC_CONFIG_SRCDIR([sample.c])
 AC_CONFIG_HEADER([config.h])
+AM_INIT_AUTOMAKE
 
 # LT Version numbers, remember to change them just *before* a release.
 #   (Interfaces removed:    CURRENT++, AGE=0, REVISION=0)
@@ -32,16 +33,16 @@
 AC_MSG_CHECKING([version script options])
 case "$host" in
 	*-*-linux*)
-	LIBSSH_VERS="$LIBSSH_LDFLAGS -Wl,--version-script,libssh.vers"
+	LIBSSH_VERS="$LIBSSH_LDFLAGS -Wl,--version-script,\$(srcdir)/libssh.vers"
 	;;
 	*-*-gnu*)
-	LIBSSH_VERS="$LIBSSH_LDFLAGS -Wl,--version-script,libssh.vers"
+	LIBSSH_VERS="$LIBSSH_LDFLAGS -Wl,--version-script,\$(srcdir)/libssh.vers"
 	;;
 esac
 AC_ARG_WITH([versioned-symbol],
 	AC_HELP_STRING([--with-versioned-symbol],[Use versioned symbols]),
 	[if test "$withval" = "yes"; then
-	   LIBSSH_VERS="$LIBSSH_LDFLAGS -Wl,--version-script,libssh.vers"
+	   LIBSSH_VERS="$LIBSSH_LDFLAGS -Wl,--version-script,\$(srcdir)/libssh.vers"
 	 else
 	   LIBSSH_VERS=""
 	 fi], [ : ])
@@ -79,7 +80,7 @@
 
 AC_CHECK_LIB([z], [deflateInit_])
 AC_SEARCH_LIBS([hstrerror],[nsl resolv])
-AC_SEARCH_LIBS([getaddrinfo],[nsl socket])
+AC_SEARCH_LIBS([getaddrinfo],[nsl socket getaddrinfo])
 AC_SEARCH_LIBS([gethostbyname],[nsl resolv])
 
 # Checks for header files.
@@ -123,5 +124,7 @@
 cfmakeraw select socket strchr strdup strerror strstr poll])
 
 AC_CONFIG_FILES([Makefile
+                 include/Makefile
+                 include/libssh/Makefile
                  libssh/Makefile])
 AC_OUTPUT
--- origsrc/libssh-0.2/include/Makefile.am	1969-12-31 18:00:00.000000000 -0600
+++ src/libssh-0.2/include/Makefile.am	2007-01-12 00:06:15.562500000 -0600
@@ -0,0 +1 @@
+SUBDIRS = libssh
--- origsrc/libssh-0.2/include/libssh/Makefile.am	1969-12-31 18:00:00.000000000 -0600
+++ src/libssh-0.2/include/libssh/Makefile.am	2007-01-12 00:06:15.562500000 -0600
@@ -0,0 +1,4 @@
+pkginclude_HEADERS = crypto.h libssh.h server.h sftp.h
+
+noinst_HEADERS = priv.h ssh1.h ssh2.h
+
--- origsrc/libssh-0.2/libssh/Makefile.am	1969-12-31 18:00:00.000000000 -0600
+++ src/libssh-0.2/libssh/Makefile.am	2007-01-12 00:06:15.578125000 -0600
@@ -0,0 +1,39 @@
+INCLUDES = -Wall -I$(top_srcdir)/include
+
+lib_LTLIBRARIES = libssh.la
+
+libssh_la_SOURCES = \
+	auth.c \
+	auth1.c \
+	base64.c \
+	buffer.c \
+	channels.c \
+	channels1.c \
+	client.c \
+	connect.c \
+	crc32.c \
+	crypt.c \
+	dh.c \
+	error.c \
+	gcrypt_missing.c \
+	gzip.c \
+	init.c \
+	kex.c \
+	keyfiles.c \
+	keys.c \
+	messages.c \
+	misc.c \
+	options.c \
+	packet.c \
+	server.c \
+	session.c \
+	sftp.c \
+	sftpserver.c \
+	string.c \
+	wrapper.c
+
+libssh_la_LDFLAGS = -export-dynamic -no-undefined \
+	-version-info $(LIBSSH_CURRENT):$(LIBSSH_REVISION):$(LIBSSH_AGE) \
+	$(LIBSSH_LDFLAGS) $(LIBSSH_VERS)
+
+EXTRA_DIST = libssh.vers
--- origsrc/libssh-0.2/libssh/connect.c	2006-12-29 09:03:59.000000000 -0600
+++ src/libssh-0.2/libssh/connect.c	2007-01-12 00:42:39.984375000 -0600
@@ -32,6 +32,9 @@
 #include <sys/time.h>
 #include <netinet/in.h>
 #include <fcntl.h>
+#ifdef __CYGWIN__
+#include <getaddrinfo.h>
+#endif
 #include "libssh/priv.h"
 
 #ifndef HAVE_SELECT
--- origsrc/libssh-0.2/libssh/crypt.c	2006-12-29 09:03:59.000000000 -0600
+++ src/libssh-0.2/libssh/crypt.c	2007-01-12 00:35:31.328125000 -0600
@@ -44,7 +44,7 @@
     return ntohl(decrypted);
 }
     
-int packet_decrypt(SSH_SESSION *session, void *data,u32 len){
+int packet_decrypt(SSH_SESSION *session, void *data,unsigned int len){
     struct crypto_struct *crypto=session->current_crypto->in_cipher;
     char *out=malloc(len);
     ssh_say(3,"Decrypting %d bytes data\n",len);
@@ -61,7 +61,7 @@
     return 0;
 }
     
-unsigned char * packet_encrypt(SSH_SESSION *session,void *data,u32 len){
+unsigned char * packet_encrypt(SSH_SESSION *session,void *data,unsigned int len){
     struct crypto_struct *crypto;
     HMACCTX ctx;
     char *out;
--- origsrc/libssh-0.2/libssh/server.c	2006-12-29 09:03:59.000000000 -0600
+++ src/libssh-0.2/libssh/server.c	2007-01-12 00:38:13.781250000 -0600
@@ -27,6 +27,7 @@
 #include <stdio.h>
 #include <sys/types.h>
 #include <sys/socket.h>
+#include <netinet/in.h>
 #include <netdb.h>
 #include <errno.h>
 #include <string.h>
--- origsrc/libssh-0.2/samplesshd.c	2006-12-29 09:03:59.000000000 -0600
+++ src/libssh-0.2/samplesshd.c	2007-01-12 00:46:48.406250000 -0600
@@ -44,8 +44,8 @@
     int sftp=0;
     int i;
     ssh_options_getopt(options,&argc,argv);
-    ssh_options_set_dsa_server_key(options,"/etc/ssh/ssh_host_dsa_key");
-    ssh_options_set_rsa_server_key(options,"/etc/ssh/ssh_host_rsa_key");
+    ssh_options_set_dsa_server_key(options,"/etc/ssh_host_dsa_key");
+    ssh_options_set_rsa_server_key(options,"/etc/ssh_host_rsa_key");
     ssh_bind=ssh_bind_new();
     ssh_bind_set_options(ssh_bind,options);
     if(ssh_bind_listen(ssh_bind)<0){
